website/demos/ColumnsReordering.tsx

```diff
@@ -29,72 +28,81 @@ function createRows(): Row[] {
   return rows;
 }
 
-function createColumns(): Column<Row>[] {
-  return [
-    {
-      key: 'id',
-      name: 'ID',
-      width: 80
-    },
-    {
-      key: 'task',
-      name: 'Title',
-      resizable: true,
-      sortable: true
-    },
-    {
-      key: 'priority',
-      name: 'Priority',
-      resizable: true,
-      sortable: true
-    },
-    {
-      key: 'issueType',
-      name: 'Issue Type',
-      resizable: true,
-      sortable: true
-    },
-    {
-      key: 'complete',
-      name: '% Complete',
-      resizable: true,
-      sortable: true
-    }
-  ];
-}
+const columns: Column<Row>[] = [
+  {
+    key: 'id',
+    name: 'ID',
+    width: 80
+  },
+  {
+    key: 'task',
+    name: 'Title',
+    resizable: true,
+    sortable: true,
+    draggable: true
+  },
+  {
+    key: 'priority',
+    name: 'Priority',
+    resizable: true,
+    sortable: true,
+    draggable: true
+  },
+  {
+    key: 'issueType',
+    name: 'Issue Type',
+    resizable: true,
+    sortable: true,
+    draggable: true
+  },
+  {
+    key: 'complete',
+    name: '% Complete',
+    resizable: true,
+    sortable: true,
+    draggable: true
+  }
+];
 
 export default function ColumnsReordering({ direction }: Props) {
   const [rows] = useState(createRows);
-  const [columns, setColumns] = useState(createColumns);
+  const [columnsOrder, setColumnsOrder] = useState(() => {
+    const map = new Map<string, number>();
+    for (const [index, column] of columns.entries()) {
+      map.set(column.key, index);
+    }
+    return map;
+  });
```

Comment:
Why not store the order as an array of indexes?
```suggestion
  const [columnsOrder, setColumnsOrder] = useState((): readonly number[] => {
    return columns.map((column, index) => index);
  });
```
then we can do
```ts
const reorderedColumns = /* useMemo => */ columnsOrder.map((index) => columns[index])
```

Reply:
Simplified a bit but using column keys array. How would we use array of indexes?

Reply:
The array would contain indexes of the columns in the original array, i.e. the original order.
That way
`[0, 1, 2]` -> `[col A, col B, col C]`
`[1, 2, 0]` -> `[col B, col C, col A]`